### ISSUE TITLE
izonereport hangs when very large file is placed into /etc/irods

### ISSUE THREAD

**Issue by JustinKyleJames** [irods/irods#7429]
One of our partners had a very large tar file in the /etc/irods directory.  This caused izonereport to hang and ultimately fail.  Once the file was moved out of that directory izonereport passed.

Since this took a while to get to the bottom of it, we might want to detect that there is an unexpected large file in that directory and handle it.

This was on iRODS 4.3.0.

**Comment by trel** [comment]
The files in that directory are each base64-encoded to be included in the payload of the `izonereport` itself.  The assumption is that they are configuration files for the server.

Agreed, we should probably provide a threshold... perhaps in `advanced_settings`... with a default of say... 5MB?

**Comment by korydraughn** [comment]
Is it possible to solve this without imposing a limit on the file size?

**Comment by trel** [comment]
well... it feels like a file that big is a mistake.    and if we don't 'skip' big files... then... what do we do?   base64 23GB files and send them as text back to the caller... in a week?

**Comment by alanking** [comment]
One alternative would be to control *which* files are being processed on a per-server basis. This could take the form of an allow list and/or deny list, a regular expression, or some combination of these (possibly even along with an optional file size limit).

**Comment by trel** [comment]
agreed that's an alternative... but... just skipping based on size seems easier to understand... and also, easier to ship a default for.
